apps/server/src/adapters/WebsocketAdapter.ts

```diff
@@ -81,29 +95,60 @@ export class SocketServer implements IAdapter {
           const { type, payload } = message;
 
           if (type === 'get-client-name') {
-            ws.send(
-              JSON.stringify({
-                type: 'client-name',
-                payload: clientId,
-              }),
-            );
+            nameWsClient(ws, this.clients.get(clientId).name);
+            return;
+          }
+
+          if (type === 'set-client-type') {
+            if (payload) {
+              const previousData = this.clients.get(clientId);
+              previousData.type = payload;
+              this.clients.set(clientId, previousData);
+            }
+            this.sendClientList();
             return;
           }
 
           if (type === 'set-client-name') {
             if (payload) {
-              const previousName = clientId;
-              clientId = payload;
-              this.clientIds.delete(previousName);
-              this.clientIds.add(clientId);
-              logger.info(LogOrigin.Client, `Client ${previousName} renamed to ${clientId}`);
+              const previousData = this.clients.get(clientId);
+              logger.info(LogOrigin.Client, `Client ${previousData.name} renamed to ${payload}`);
+              previousData.name = payload;
+              this.clients.set(clientId, previousData);
+              nameWsClient(ws, this.clients.get(clientId).name);
+            }
+            this.sendClientList();
+            return;
+          }
+
+          if (type === 'set-client-identify') {
+            if (payload) {
+              const targetClient = this.clients.get(payload.target);
+              this.clients.set(payload.target, { ...targetClient, identify: payload.state });
+              this.sendClientList();
+            }
+            return;
+          }
+
+          if (type === 'set-client-redirect') {
+            if (payload && this.clients.has(payload.target)) {
+              this.sendAsJson({ type: 'client-redirect', payload });
             }
-            ws.send(
-              JSON.stringify({
-                type: 'client-name',
-                payload: clientId,
-              }),
-            );
+            return;
+          }
+
+          if (type === 'set-client-rename') {
+            if (payload && this.clients.has(payload.target)) {
+              const targetID = payload.target;
+              const newName = payload.name;
```

Comment:
question: can we assume that payload being truthy means it carries the correct values?
What do we do in the rest of the app? would it be worth wrapping this in a try-catch?